hudi-common/src/main/java/org/apache/hudi/metadata/BaseTableMetadata.java

```diff
@@ -138,13 +144,17 @@ public FileStatus[] getAllFilesInPartition(Path partitionPath)
       }
     }
 
-    return new FileSystemBackedTableMetadata(getEngineContext(), hadoopConf, dataBasePath, metadataConfig.shouldAssumeDatePartitioning())
+    return new FileSystemBackedTableMetadata(getEngineContext(), hadoopConf, dataBasePath.toString(), metadataConfig.shouldAssumeDatePartitioning())
         .getAllFilesInPartition(partitionPath);
   }
 
   @Override
   public Map<String, FileStatus[]> getAllFilesInPartitions(List<String> partitions)
       throws IOException {
+    if (partitions.isEmpty()) {
+      return Collections.emptyMap();
```

Comment:
If `BaseHoodieTableFileIndex#getAllPartitionPathsUnchecked` returns `Collections.singletonList("")` then should we add an entry for `""` in the map, or rather make `getAllPartitionPathsUnchecked` return `Collections.emptyList()`